platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,437 @@
+# TSS - Roster Proposal
+
+---
+
+## Summary
+
+This proposal outlines the design specification of rosters to support Threshold Signature Scheme (TSS) block signing.
+It details the roster's lifecycle, data storage, API, and necessary changes to associated components within the
+platform.
+
+| Metadata           | Entities                                      | 
+|--------------------|-----------------------------------------------|
+| Designers          | Kore, Anthony, Ed, Cody, Austin               |
+| Functional Impacts | TSS, Genesis, Cryptography, Reconnect         |
+| Related Proposals  | TSS-Library, TSS-Ledger-Id, TSS-Block-Signing |
+| HIPS               | N/A                                           |
+
+## Purpose and Context
+
+The introduction of the Threshold Signature Scheme (TSS) requires a new mechanism for managing node participation in
+consensus and block signing.
+The Roster, an immutable subset of the address book, will provide this mechanism, ensuring efficient and secure key
+management for TSS operations.
+This proposal provides a specification for the behavior of rosters starting from their creation from the
+Candidate Address Book (CAB) to their terminal state within the TSS specification.
+A roster reaches a terminal state when it is either adopted by the platform or replaced by a new roster.
+
+The Future Address Book (FAB) is a dynamic list maintained within the App that reflects the desired future state of the
+network's nodes. It will be continuously updated by HAPI transactions, such as those that create, update, or delete
+nodes.
+The CAB is a snapshot of the Future Address Book (FAB), and it is created when the app decides to adopt a new address
```

Comment:
It is unlikely that something like a Candidate Address Book or Future Address Book will ever exist. Please refrain from using these terms because they are misleading.

Reply:
Thanks @netopyr. Reworded & updated diagram as discussed offline.